clients/src/main/java/org/apache/kafka/common/security/auth/AuthenticationContext.java

```diff
@@ -16,10 +16,21 @@
  */
 package org.apache.kafka.common.security.auth;
 
+import java.net.InetAddress;
+
 /**
  * An object representing contextual information from the authentication session. See
  * {@link SaslAuthenticationContext} and {@link SslAuthenticationContext}.
  */
 public interface AuthenticationContext {
+    /**
+     * Underlying security protocol of the authentication session.
+     * @return The name of the security protocol (i.e. PLAINTEXT, SASL_PLAINTEXT, SASL_SSL, SSL)
+     */
+    String securityProtocol();
```

Comment:
I'd call this `securityProtocolName` so that we have the option of exposing the `SecurityProtocol` type in the future.